### DISCUSSION TITLE
Custom Tabs on Android

### DISCUSSION THREAD

**Discussion by MatthiasKroiss** [supabase-community/supabase-kt#320] [Q&A]
In my Android app I'm trying to use the custom tabs feature of gotrue, but it doesn't work i always get the following exception
``android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
``
This is my code to initialize my supabase client:
``package com.aiui.paperui

import android.app.Application
import android.content.Context
import io.github.jan.supabase.SupabaseClient
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.gotrue.ExternalAuthAction
import io.github.jan.supabase.gotrue.GoTrue
import io.github.jan.supabase.gotrue.gotrue
import io.github.jan.supabase.postgrest.Postgrest
import kotlinx.coroutines.runBlocking

class PaperUI : Application() {

    // Singleton for the Application Class
    object instance {
        lateinit var supabaseClient: SupabaseClient

    }

    override fun onCreate() {
        super.onCreate()

            // Initialize Supabase Client
            runBlocking {
                instance.supabaseClient = initSupabaseClient()
                instance.supabaseClient.gotrue.loadFromStorage(autoRefresh = true)
            }

    }

    fun initSupabaseClient(): SupabaseClient {
        val client = createSupabaseClient(
            supabaseUrl = com.aiui.paperui.BuildConfig.SUPABASE_URL,
            supabaseKey = com.aiui.paperui.BuildConfig.SUPABASE_ANON_KEY
        ) {
            install(Postgrest)
            install(GoTrue) {
                scheme = "aiui.paperui"
                host = "auth"
                // not working yet
                defaultExternalAuthAction = ExternalAuthAction.CUSTOM_TABS
            }
        }

        return client
    }

}
`` 

And this the code to login through github
``fun loginWithGithub() {
        viewModelScope.launch {
            _isLoading.value = true
                PaperUI.instance.supabaseClient.gotrue.loginWith(Github)
        }
    }
``

Please help me with this issue

**Comment by jan-tennert** [comment]
> FLAG_ACTIVITY_NEW_TASK

Hey! This seems to be a bug, I'm publishing `1.4.6-dev-1` with a possible fix (available in around ~30min), can you try it and tell me if it works?

**Reply by MatthiasKroiss** [reply]
Thanks, now it's working in custom tabs except that it wants to open another browser if i click on sign in at the github page but that's an issue of github i think.

**Reply by MatthiasKroiss** [reply]
News! I dont have any details yet but i got reviews that on Android 14 the solution doesn't work and the app crashes. I'm trying to provide an erro stack in the next days

**Reply by jan-tennert** [reply]
Weird, let me know if you have something!

**Reply by MatthiasKroiss** [reply]
Seems to be bug after updating to Android 14 reinstall of the app helped.
